packages/openapi3/src/examples.ts

```diff
@@ -231,6 +264,352 @@ export function getBodyValue(value: Value, properties: HttpProperty[]): Value |
+      string -> "blue"
+      array -> ["blue", "black", "brown"]
+      object -> { "R": 100, "G": 200, "B": 150 }
+
+    | explode | string        | array                               | object                   |
+    | ------- | ------------- | ----------------------------------- | ------------------------ |
+    | false   | ;color=blue   | ;color=blue,black,brown             | ;color=R,100,G,200,B,150 |
+    | true    | ;color=blue   | ;color=blue;color=black;color=brown | ;R=100;G=200;B=150       |
+  */
+
+  const joiner = explode ? ";" : ",";
+  const prefix = explode ? "" : `${name}=`;
+  if (tk.value.isArray(originalValue)) {
+    const pairs: string[] = [];
+    for (const value of originalValue.values) {
+      if (!isSerializableScalarValue(value)) continue;
+      pairs.push(explode ? `${name}=${value.value}` : `${value.value}`);
+    }
+    return tk.value.createString(`;${prefix}${pairs.join(joiner)}`);
+  }
+
+  if (tk.value.isObject(originalValue)) {
+    const sep = explode ? "=" : ",";
+    const pairs: string[] = [];
+    for (const [key, { value }] of originalValue.properties) {
+      if (!isSerializableScalarValue(value)) continue;
+      pairs.push(`${key}${sep}${value.value}`);
+    }
+    return tk.value.createString(`;${prefix}${pairs.join(joiner)}`);
+  }
+
+  if (tk.value.isNull(originalValue)) {
+    return tk.value.createString(`;${name}`);
+  }
+
+  if (isSerializableScalarValue(originalValue)) {
+    return tk.value.createString(`;${name}=${originalValue.value}`);
+  }
+
+  return;
+}
+
+function getParameterDelimitedValue(
+  program: Program,
+  originalValue: Value,
+  property: Extract<HttpParameterProperties, { kind: "query" }>,
+  delimiter: " " | "|",
+): Value | undefined {
+  const { explode, name } = property.options;
+  // Serialization is undefined for explode=true
+  if (explode) return undefined;
+
+  const tk = $(program);
+  /*
+    https://spec.openapis.org/oas/v3.0.4.html#style-examples
+      array -> ["blue", "black", "brown"]
+      object -> { "R": 100, "G": 200, "B": 150 }
+
+    | style | explode | string | array                       | object                             |
+    | ----- | ------- | ------ | --------------------------- | ---------------------------------- |
+    | pipe  | false   | n/a    | ?color=blue%7Cblack%7Cbrown | ?color=R%7C100%7CG%7C200%7CB%7C150 |
+    | pipe  | true    | n/a    | n/a                         | n/a                                |
+    | space | false   | n/a    | ?color=blue%20black%20brown | ?color=R%20100%20G%20200%20B%20150 |
+    | space | true    | n/a    | n/a                         | n/a                                |
+  */
+
+  if (tk.value.isArray(originalValue)) {
+    const pairs: string[] = [];
+    for (const value of originalValue.values) {
+      if (!isSerializableScalarValue(value)) continue;
+      pairs.push(`${value.value}`);
+    }
+    return tk.value.createString(`?${name}=${encodeURIComponent(pairs.join(delimiter))}`);
```

Comment:
I need to double check if there's anywhere else I need to encode URI components.